iroh-bytes/src/store/file/validate.rs

```diff
@@ -0,0 +1,469 @@
+                            continue;
+                        };
+                        let hash = Hash::from(hash);
+                        if !entries.contains(&hash) {
+                            orphaned_data.insert(hash);
+                            entry_warn!(hash, "orphaned data file");
+                        }
+                    }
+                    None => {
+                        warn!("unexpected data file in data directory: {}", path.display());
+                    }
+                },
+                Some("obao4") => match path.file_stem().and_then(|x| x.to_str()) {
+                    Some(stem) => {
+                        let mut hash = [0u8; 32];
+                        let Ok(_) = hex::decode_to_slice(stem, &mut hash) else {
+                            warn!(
+                                "unexpected outboard file in data directory: {}",
+                                path.display()
+                            );
+                            continue;
+                        };
+                        let hash = Hash::from(hash);
+                        if !entries.contains(&hash) {
+                            orphaned_outboardard.insert(hash);
+                            entry_warn!(hash, "orphaned outboard file");
+                        }
+                    }
+                    None => {
+                        warn!(
+                            "unexpected outboard file in data directory: {}",
+                            path.display()
+                        );
+                    }
+                },
+                Some("sizes4") => match path.file_stem().and_then(|x| x.to_str()) {
+                    Some(stem) => {
+                        let mut hash = [0u8; 32];
+                        let Ok(_) = hex::decode_to_slice(stem, &mut hash) else {
+                            warn!(
+                                "unexpected outboard file in data directory: {}",
+                                path.display()
+                            );
+                            continue;
+                        };
+                        let hash = Hash::from(hash);
+                        if !entries.contains(&hash) {
+                            orphaned_sizes.insert(hash);
+                            entry_warn!(hash, "orphaned outboard file");
+                        }
+                    }
+                    None => {
+                        warn!(
+                            "unexpected outboard file in data directory: {}",
+                            path.display()
+                        );
+                    }
+                },
+                _ => {
+                    warn!("unexpected file in data directory: {}", path.display());
+                }
+            }
+        }
+        if repair {
+            info!("repairing - removing orphaned files and inline data");
```

Comment:
would be nice to print some stats at the end, like "removed 18 orphaned files"

Reply:
Yeah, added that in the batch branch.